homeassistant/components/nuki/binary_sensor.py

```diff
@@ -70,3 +76,35 @@ def door_sensor_state_name(self):
     def is_on(self):
         """Return true if the door is open."""
         return self.door_sensor_state == STATE_DOORSENSOR_OPENED
+
+
+class NukiRingactionEntity(NukiEntity[NukiDevice], BinarySensorEntity):
+    """Representation of a Nuki Opener Ringaction."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_icon = "mdi:bell-ring"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self._nuki_device.nuki_id}_ringaction"
+
+    @property
+    def extra_state_attributes(self):
+        """Return the device specific state attributes."""
+        data = {
+            ATTR_NUKI_ID: self._nuki_device.nuki_id,
+            ATTR_RING_TIMESTAMP: self._nuki_device.ring_action_timestamp,
+        }
+        return data
+
+    @property
+    def ring_action_state(self):
+        """Return the state of the ring action."""
+        return self._nuki_device.ring_action_state
+
+    @property
+    def is_on(self) -> bool:
+        """Return the value of the ring action state."""
+        return self.ring_action_state
```

Comment:
why the extra property and not return `self._nuki_device.ring_action_state` directly?

Reply:
thank you, I changed it